### PR TITLE
Import fmtlib/fmt@6.2.0, a C++20-style format library

### PR DESCRIPTION
**Pull request by DHowett**
We received a request from our localization team to switch from
printf-style format strings (%s, %u) to format strings with positional
argument support. I've been hoping for a long time to take a dependency
on C++20's std::format, but we're just not somewhere we can do that.
Enter fmt. fmt is _exactly_ the library we need.

Minor comparison:

### then

```c++
std::wstring_view world = /* ... */;
auto str{ wil::str_printf<std::wstring>(L"hello %.*s",
	   gsl::narrow_cast<size_t>(world.size()),
	   world.data()) };
```

### now

```c++
auto str{ fmt::format(L"hello {0}", world) };
```

If you really want to use the print specifiers:

```
auto str{ fmt::printf(L"hello %s", world) };
```

It's got optional compile-time checking for format strings and is
MIT-licensed. Eventually, we should be able to replace fmt:: with std::
and end up pretty much where we left off.
What more could you ask for?

It has a header-only mode, but it seemed easier on the compiler (and the
linker) to throw together a quick static library project that produces
the source-level components of fmt.

### CODE REVIEW

**Comment by DHowett** [comment]
Also? It passes static analysis (!) on its own (!)

**Comment by DHowett** [comment]
For heck's sake, I used the wrong GitHub account. Oh well, it'll be fine.

File: OpenConsole.sln [hunk 1]
```diff
@@ -300,6 +300,10 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "winconpty.Tests.Feature", "
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "TerminalAzBridge", "src\cascadia\TerminalAzBridge\TerminalAzBridge.vcxproj", "{067F0A06-FCB7-472C-96E9-B03B54E8E18D}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "_dep", "_dep", "{6FFCC3B4-A6CB-44FF-BFE3-1EF02CCD7557}"
```
Comment (zadjii-msft) [inline OpenConsole.sln:303]: I thought we already had a top-level "_Dependencies" folder in the solution?

Comment (DHowett-MSFT) [reply OpenConsole.sln:303]: That one doesn’t seem to contain any buildable projects. I’m happy to move this in there, but /shrug!

**Comment by DHowett-MSFT** [comment]
I’d prefer to make this an atomic import

**Comment by DHowett-MSFT** [comment]
@msftbot make sure @miniksa signs off on this

Michael, made a call to introduce `src/dep` which _only contains project files_. Those project files point out to source files in the actual dep folder.

**Comment by miniksa** [comment]
> 
> 
> @msftbot make sure @miniksa signs off on this
> 
> Michael, made a call to introduce `src/dep` which _only contains project files_. Those project files point out to source files in the actual dep folder.

Yeah, sure. It makes some semblance of sense to me. Kinda like `tools` vs `src/tools`.

File: OpenConsole.sln (hunk 1 above)
Comment (miniksa) [reply OpenConsole.sln:303]: I'd rather you put it in the existing one, in this case.

**Comment by miniksa** [comment]
You aren't putting it into `LibraryIncludes.h`? Or referencing it anywhere?

**Comment by DHowett-MSFT** [comment]
> You aren't putting it into `LibraryIncludes.h`? Or referencing it anywhere?

Correct. Straight import and build right now. I'm wary of adding it to the global include path until we need it in more than one project.

Right now, the only place we _require_ positionals is in TermConnection. I'd like to understand where in conhost _can_ and then _should_ use it before opening it up to conhost b/c of the OS vpack problem

**Comment by miniksa** [comment]
> 
> 
> > You aren't putting it into `LibraryIncludes.h`? Or referencing it anywhere?
> 
> Correct. Straight import and build right now. I'm wary of adding it to the global include path until we need it in more than one project.
> 
> Right now, the only place we _require_ positionals is in TermConnection. I'd like to understand where in conhost _can_ and then _should_ use it before opening it up to conhost b/c of the OS vpack problem

Nnnn ok. I greatly prefer to just have support libraries like this just "be accessible" through the global `LibraryIncludes.h` but I understand the limitations of the situation. We should fix the vpack problem soon as we can, though.

**Comment by DHowett-MSFT** [comment]
Okay, decided to throw it into LibraryIncludes and link it from ConTypes.

**Pull request merged**